index.js

```diff
@@ -19,17 +20,40 @@ module.exports.env = function() {
 };
 
 module.exports.os = function() {
-  if (typeof os.userInfo === 'function') {
-    return os.userInfo().username;
+  try {
+    if (typeof os.userInfo === 'function') {
+      return os.userInfo().username;
+    }
+  } catch(e) {
+    handleUserInfoError(e);
+    return;
   }
 };
 
 module.exports.execSync = function() {
-  var username = require('child_process').execSync('whoami').toString().trim();
+  try {
+    var username = require('child_process').execSync('whoami').toString().trim();
 
-  if (process.platform === 'win32') {
-    username = username.replace(/^.*\\/, ''); // remove DOMAIN stuff
-  }
+    if (username) {
+      if (process.platform === 'win32') {
+        return username = username.replace(/^.*\\/, ''); // remove DOMAIN stuff
+      }
+
+      return username;
+    }
 
-  return username;
+    return require('child_process').execSync('id', ['-un']).toString().trim();
+  } catch(e) {
+    handleUserInfoError(e);
+    return;
+  }
 };
+
+module.exports.handleUserInfoError = function(e) {
```

Comment:
LGTM

Reply:
Ooops - actually I did not mean to export that. I only did that for testing. I think it should be a private function.